### PR TITLE
CF proxy wrapper

### PR DESCRIPTION
**Pull request by KirilKabakchiev**

### CODE REVIEW

File: platform/config.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+		clientConfig.ApiAddress = platformSettings.Cf.Api
+	}
+	if len(platformSettings.Cf.ClientID) != 0 {
+		clientConfig.ClientID = platformSettings.Cf.ClientID
+	}
+	if len(platformSettings.Cf.ClientSecret) != 0 {
+		clientConfig.ClientSecret = platformSettings.Cf.ClientSecret
+	}
+	if len(platformSettings.Cf.Username) != 0 {
+		clientConfig.ClientID = platformSettings.Cf.ClientID
```
Comment (dotchev) [inline platform/config.go]: probably `Cf.Username`

File: platform/config.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	}
+	if len(platformSettings.Cf.Password) != 0 {
+		clientConfig.ClientSecret = platformSettings.Cf.ClientSecret
```
Comment (dotchev) [inline platform/config.go]: probably `Cf.Password`

File: platform/client.go [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+	"github.com/cloudfoundry-community/go-cfclient"
+	"github.com/sirupsen/logrus"
+)
+
+type PlatformClient struct {
+	cfClient *cfclient.Client
+	reg      *RegistrationDetails
+}
+
+var _ platform.Client = &PlatformClient{}
```
Comment (pankrator) [inline platform/client.go]: It took me some time to understand where this `platform.Client` comes from. May be use some alias for the imported platform. Currently I don't have better idea for name :(

File: platform/config.go [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+type PlatformClientConfiguration struct {
+	*cfclient.Config
+
+	Reg        *RegistrationDetails
+	createFunc func(*cfclient.Config) (*cfclient.Client, error)
+}
+
+var _ platform.ClientConfiguration = &PlatformClientConfiguration{}
+
+func (c *PlatformClientConfiguration) CreateFunc() (platform.Client, error) {
```
Comment (pankrator) [inline platform/config.go]: Why this method is `CreateFunc`, there is another one on the same structure starting with lower case. It is hard to understand the code.

Comment (KirilKabakchiev) [reply platform/config.go]: I think that the createFunv (lower case) is leftover from refactoring

File: platform/client.go [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+		}
+		clientBrokers = append(clientBrokers, serviceBroker)
+	}
+	return clientBrokers, nil
+}
+
+func (b PlatformClient) CreateBroker(r *platform.CreateServiceBrokerRequest) (*platform.ServiceBroker, error) {
+	logrus.Debugf("Creating broker with name [%s]...", r.Name)
+	logrus.Debug("Obtaining CF API access token...")
+	_, err := b.cfClient.GetToken()
```
Comment (dpanayotov) [inline platform/client.go]: this is not needed, cf client takes care of this

File: main.go [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+	if err != nil {
+		logrus.Fatal("Error loading configuration: ", err)
+	}
+
+	cfConfig, err := platform.DefaultConfig()
+	if err != nil {
+		logrus.Fatal("Error loading configuration: ", err)
+	}
+
+	cfg, err := config.New(sbproxyConfig, osbConfig, smConfig, cfConfig)
```
[the remark this answers is not in the document]
Comment (dotchev) [reply main.go]: osb-lib & osb-client are implementation details of the sb-proxy core lib, they should be visible to its users
conceptually the sb-lib should take two pieces of configuration - where it receives OSB calls & where it dispatches those calls (SM), additionally it should take a platform client implementation to handle SB reg, unreg & update
a platform-specific wrapper should not pass its own platform config to the sb-proxy lib, it should pass directly the client interface

**Pull request merged**